component/core/src/main/java/io/meeds/social/space/service/SpaceServiceImpl.java

```diff
@@ -1405,4 +1408,86 @@ private String buildPrettyName(String... names) {
             .orElseThrow();
   }
 
+  private void checkSubspaceCreationAllowed(Space space, long parentSpaceId) throws SpaceException {
+    if (parentSpaceId <= 0) {
+      return;
+    }
+    Space parentSpace = getSpaceById(parentSpaceId);
+    SpaceTemplate parentTemplate = spaceTemplateService.getSpaceTemplate(parentSpace.getTemplateId());
+    if (parentTemplate == null) {
+      throw new SpaceException(Code.UNKNOWN_SPACE_TEMPLATE,
+                               String.format("Unknown parent space template for space %s", parentSpace.getDisplayName()));
+    }
+
+    Integer subspacesMaxLimit = parentTemplate.getSubspacesMaxLimit();
+    List<String> allowedSubspaceTemplates = parentTemplate.getAllowedSubspaceTemplates();
+
+    SpaceFilter spaceFilter = new SpaceFilter();
+    spaceFilter.setParentSpaceId(parentSpaceId);
+    ListAccess<Space> existingSubspacesAccess = getAllSpacesByFilter(spaceFilter);
+    Space[] existingSubspaces;
+    try {
+      existingSubspaces = existingSubspacesAccess.load(0, -1);
+    } catch (Exception e) {
+      throw new SpaceException(Code.ERROR_DATASTORE, "Failed to load subspaces for parent " + parentSpaceId, e);
+    }
+
+    int totalSubspaces = existingSubspaces != null ? existingSubspaces.length : 0;
+
+    if (subspacesMaxLimit != null && subspacesMaxLimit != 0 && totalSubspaces >= subspacesMaxLimit) {
+      throw new SpaceException(Code.SUBSPACES_LIMIT_REACHED,
+                               String.format("Cannot create more subspaces under '%s' (max %d reached)",
+                                             parentSpace.getDisplayName(),
+                                             subspacesMaxLimit));
+    }
+
+    if (allowedSubspaceTemplates == null || allowedSubspaceTemplates.isEmpty()) {
+      return;
+    }
+
+    long templateId = space.getTemplateId();
+    String matchedRule = null;
+    for (String rule : allowedSubspaceTemplates) {
+      if (rule != null && rule.startsWith(templateId + ":")) {
```

Comment:
Add the format check here before assigning the `matchedRule`, this way you assign only a valid rule

```suggestion
      if (rule != null && rule.startsWith(templateId + ":") && matchedRule.split(":").length == 2) {
```